benchmarks/PNP/pnp_EAFE.cpp

```diff
@@ -31,7 +31,38 @@ double upper_cation_val = 1.0;  // 1 / m^3
 double lower_anion_val = 1.0;  // 1 / m^3
 double upper_anion_val = 0.1;  // 1 / m^3
 double lower_potential_val = -1.0;  // V
-double upper_potential_val = 1.0;  // V
+double upper_potential_val = 1.0;  //
+
+void update_solution_pnp(dolfin::Function& iterate0, dolfin::Function& iterate1, dolfin::Function& iterate2, \
+    dolfin::Function& update, double& relative_residual, pnp_and_source::LinearForm & L, const dolfin::DirichletBC bc, dolfin::EigenVector& b, double DampFactor, int maxDamp)
+{
+    double new_residual=0.0;
+    int i=1;
+    dolfin::Function _iterate0(iterate0); dolfin::Function _iterate1(iterate1); dolfin::Function _iterate2(iterate2);
+    dolfin::Function update0(update[0]); dolfin::Function update1(update[1]); dolfin::Function update2(update[2]);
+    *(_iterate0.vector()) += *(update0.vector()); *(_iterate1.vector()) += *(update1.vector()); *(_iterate2.vector()) += *(update2.vector());
+    L.CatCat = _iterate0;
+    L.AnAn   = _iterate1;
+    L.EsEs   = _iterate2;
+    assemble(b, L);
+    bc.apply(b);
+    new_residual = b.norm("l2");
+    while ( (new_residual>relative_residual) and (i<maxDamp) )
+    {
+      *(_iterate0.vector()) = *(iterate0.vector()); *(_iterate1.vector()) = *(iterate1.vector()); *(_iterate2.vector()) = *(iterate2.vector());
+      *(update0.vector()) *= DampFactor; *(update1.vector()) *= DampFactor; *(update2.vector()) *= DampFactor;
+      *(_iterate0.vector()) += *(update0.vector()); *(_iterate1.vector()) += *(update1.vector()); *(_iterate2.vector()) += *(update2.vector());
+      i++;
+      L.CatCat = _iterate0;
+      L.AnAn   = _iterate1;
+      L.EsEs   = _iterate2;
+      assemble(b, L);
+      bc.apply(b);
+      new_residual = b.norm("l2");
+    }
+    *(iterate0.vector()) = *(_iterate0.vector()); *(iterate1.vector()) = *(_iterate1.vector()); *(iterate2.vector()) = *(_iterate2.vector());
+    relative_residual = new_residual;
```

Comment:
``new_residual`` is not the relative_residual.  Isn't that what we should be computing here?  It looks like that's what should be returned